django/db/migrations/state.py

```diff
@@ -117,10 +117,7 @@ def _find_reload_model(self, app_label, model_name, delay=False):
         else:
             # Get all relations to and from the old model before reloading,
             # as _meta.apps may change
-            if delay:
-                related_models = get_related_models_tuples(old_model)
-            else:
-                related_models = get_related_models_recursive(old_model)
+            related_models = get_related_models_recursive(old_model)
```

Comment:
Instead of generally looking for related models here regardless of the value for `delay` deceives the purpose of `delay=True`. I think it's up to `AlterField` and `RenameField` to determine if the field is used in a ForeignKey constrained or is a PrimaryKey.

Reply:
I.e. https://github.com/timgraham/django/blob/57a41a005053ca2caf877514ae9f89b98c49dbdc/django/db/migrations/operations/fields.py#L200 should include a check for `field.primary_key`

Reply:
Right, except it's more complicated because of `ForeignKey.to_field`.